src/Tutorial.ts

```diff
@@ -0,0 +1,438 @@
+
+    constructor(name: string,description: string, tutorialSteps: TutorialStep[]){
+        this.name = name;
+        this.description = description;
+        this.tutorialSteps = tutorialSteps;
+
+    }
+
+
+    getTutorialSteps = () : TutorialStep[] => {
+        return this.tutorialSteps;
+    }
+
+    getName = () : string => {
+        return this.name;
+    }
+
+    getDescription = () : string => {
+        return this.description;
+    }
+
+    initiateTutorial = (tutorialName:string) : void => {
+        const that = this;
+
+        Eagle.tutorials.forEach(function(tut){
+            if(tutorialName === tut.getName()){
+                //this is the requsted tutorial
+                activeTut = tut
+                activeTutStepsNo = tut.getTutorialSteps().length
+                activeTutCurrentStep = 1
+                that.initiateTutStep('next')
+            }
+        })
+        this.initiateTutQuickSelect()
+    }
+
+    initiateTutQuickSelect = () : void => {
+        var that = this
+        $("body").on('keydown.tutEventListener',function(e){
+
+            switch(e.which) {
+                case 37: // left
+                    that.tutButtonPrev()
+                break;
+        
+                case 38: // up
+                    that.tutButtonPrev()
+                break;
+        
+                case 39: // right
+                    if(activeTut.getTutorialSteps()[activeTutCurrentStep-1].getType() != TutorialStep.Type.Press){
+                        that.tutButtonNext() 
+                    }
+                break;
+        
+                case 40: // down
+                    that.tutButtonNext()
+                break;
+
+                case 27: //escape
+                e.preventDefault()
+                    that.tutButtonEnd()
+                break;
+        
+                default: return; // exit this handler for other keys
+            }
+        })
+    }
+
+    //cooldown function that prevents too many actions that would cause the tutorial steps to go out of whack
+    startCooldown = () : void => {
+        cooldown = true
+        setTimeout(function(){
+            cooldown = false
+        }, 500)
+    }
+
+    initiateTutStep = (direction:string) :void => {
+        const eagle = Eagle.getInstance()
+
+        var tutStep = activeTut.getTutorialSteps()[activeTutCurrentStep-1]
+
+        //if there is a preFunction set, then we execute it here
+        var preFunction
+
+        if(direction === 'next'){
+            preFunction = tutStep.getPreFunct()
+        }else if(direction === 'prev'){
+            preFunction = tutStep.getBackPreFunct()
+        }
+        
+        if(preFunction != null){
+            preFunction(eagle)
+        }
+        
+        //we always pass through the wait funciton, it is decided there if we actully wait or not
+        this.initiateWaitForElement(tutStep.getWaitType())
+    }
+
+    initiateWaitForElement = (waitType:Wait.Type) :void => {
```

Comment:
This function seems to do more than wait. It calls pickStepType(), which then initiates/runs the step.

It seems like this behaviour is a bit hidden behind the name "wait"